src/main/java/seedu/address/model/travelplanner/ModelManager.java

```diff
@@ -151,62 +150,68 @@ public void setActivity(Activity target, Activity editedActivity) {
     @Override
     public void setDirectory(Directory dir) {
         directory = dir;
-        setCurrentTravelPlan();
     }
 
     @Override
     public Directory getDirectory() {
         return directory;
     }
 
-    /**
-     * Sets the current travel plan to the current directory.
-     */
-    private void setCurrentTravelPlan() {
-        if (directory.isTravelPlan()) {
-            currentTravelPlan = (TravelPlan) directory;
-            currentFilteredTravelPlanObjectList = new FilteredList<>(currentTravelPlan.getActivityList());
-        }
-    }
+    //========== TravelPlanObjectList ===================================================
 
     @Override
     public void setToActivityList() {
-        currentFilteredTravelPlanObjectList = new FilteredList<>(currentTravelPlan.getActivityList());
+        currentFilteredTravelPlanObjectList = new FilteredList<>(directory.getActivityList());
     }
 
     @Override
     public void setToAccommodationList() {
-        currentFilteredTravelPlanObjectList = new FilteredList<>(currentTravelPlan.getAccommodationList());
+        assert directory instanceof TravelPlan
```

Comment:
Why did you use assert here btw? Cos I think assert won't be run if we didnt specify it and the user probably wont run our program with assertion on. Maybe check it using if block and throw an exception? Same for all the methods below

Reply:
It's for testing purposes and to also remind programmers under what conditions this call should be made only. I wrote this with the intention that the programmer should only call setToAccommodationList() if the directory is a Travelplan. I was thinking about try catch, but it's not a scenario the user will face, since this should be configured by programmers (and should not happen because of user input/calls). It's also okay to write try catch, but there should not be any exception in the first place unless there is a bug in the program.

Reply:
oh okok then Logic is the one that have to make sure the directory is the correct instance?

Reply:
yeap! so commands must check the directory before setting accommodationlist etc, since commands will have to know the directory to execute local commands anyway.